### DISCUSSION TITLE
How to limit query with "like" operator

### DISCUSSION THREAD

**Discussion by cntwelve** [trinodb/trino#20575] [Q&A]
Hi, everyone:

In my productive env, there's a lot of qureies to some table with "like" operator, which make the cluster heavy load. Is there any method to limit such query? For example, use access control simlar to limit function call.

thanks a lot!

---------------------------------
example:
query:
    select column_a from table_b where column_c like 'data%';
Can  I use ranger to limit queries whith "like" to table_b?

**Comment by hirolee88** [comment]
io.trino.execution.QueryPreparer#prepareQuery  where  instanceof LikePredicate

Does Ranger need customization?

**Comment by cntwelve** [comment]
Sorry for a long time offline.

(1) Yes, the code you mentioned can do the job.
(2) I check issues again, and find this link:
https://github.com/trinodb/trino/issues/19912
operator's filter can be another way
(3) And I find trino-gateway that does the LB and Proxy function, can do the statement filter. But I cannt estimate the performance.

Thx.
